datalake_api/fetcher.py

```diff
@@ -37,20 +40,22 @@ def _adjusted_path(self):
             return m.group('path')
         return self.metadata['path']
 
-    def _guess_type(self):
-        self.content_type, self.content_encoding = \
-            guess_type(self._adjusted_path)
-        if self.content_type is None and self._is_log_file():
-            self.content_type = 'text/plain'
+    @memoized_property
+    def content_type(self):
+        _type, _encoding = guess_type(self._adjusted_path)
+        return _type
+
+    @memoized_property
+    def content_encoding(self):
+        mime = magic.from_buffer(self._header, mime=True)
+        if mime == 'application/gzip':
+            return 'gzip'
+        return None
 
     _KNOWN_LOGS = ['syslog', 'dmesg']
 
-    def _is_log_file(self):
-        return self.metadata['what'] in self._KNOWN_LOGS or \
-            self._adjusted_path.endswith('.log') or \
-            self._adjusted_path.endswith('.log.gz') or \
-            self._adjusted_path.endswith('.log.1.gz') or \
-            self._adjusted_path.endswith('.log.1')
+    def read(self):
```

Comment:
Ah yes. That was my first attempt. But this is not a real file handle. It's a streaming file handle returned by boto. It does not have seek. I considered managing an offset, but then I just decided to support a pseudo read that does not take a length argument. Considering that this is how the sole caller uses it, I'm not too worried. Anyway, the behavior is buggy and I'll fix it.